Insertion_sort.c

```diff
@@ -0,0 +1,35 @@
+#include "sort.h"
+
+/**
+ * insertion_sort_list - perform insertion sort on a linked list
+ * @list: head node of the linked list
+ */
+void insertion_sort_list(listint_t **list)
+{
+	listint_t *current, *swap, *prv;
+
+	if (!list || !*list)
+		return;
```

Comment:
The variable names used in the code are concise and self-explanatory, which is good for readability.  The function checks if the list and *list are not NULL before proceeding with the sorting. This is important to avoid segmentation faults or undefined behavior.